File: app/src/main/java/com/example/kotlin/java/Father.java
package com.example.kotlin.java;

/**
 * @author zhangquan
 */
public abstract class Father {
    private String name;
    private int age;
    public int value = 1;
    private String priVar = "priVar";
    public String pubVar = "pubVar";
    public static String staticVar = "staticVar";
    private static final String fianlStaticVar="fianlStaticVar";

    public Father(String name) {
        this.name = name;
    }

    public Father(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public void setData(String name) {
        this.name = name;
    }

    public void setData(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return this.name;
    }

    public static String staticMethod(String name, int age) {
        return name;
    }

    private void doUpdate(MyInterface callback) {
        callback.update(name, age);
    }

    public void update() {
        doUpdate(new MyInterface() {
            @Override
            public void update(String name, int age) {

            }
        });
    }

    public abstract void method();

    public abstract String method2(String name);
}

File: app/src/main/java/com/example/kotlin/kotlin/Father.kt
package com.example.kotlin.kotlin

import com.example.kotlin.kotlin.interfaces.MyInterface

/**
 * @author zhangquan
 */
abstract class Father {
    private var name: String
    private var age = 0
    var value = 1
    private val priVar = "priVar"
    var pubVar = "pubVar"

    constructor(name: String) {
        this.name = name
    }

    constructor(name: String, age: Int) {
        this.name = name
        this.age = age
    }

    fun setData(name: String) {
        this.name = name
    }

    fun setData(name: String, age: Int) {
        this.name = name
        this.age = age
    }

    private fun doUpdate(callback: MyInterface) {
        callback.update(name, age)
    }

    fun update() {
        doUpdate(object : MyInterface {
            override var name: String=""

            override fun update(name: String, age: Int) {

            }
        })
    }

    abstract fun method()
    abstract fun method2(name: String): String

    /**
     * 静态变量、静态方法
     */
    companion object {
        var staticVar = "staticVar"
        private const val fianlStaticVar = "fianlStaticVar"
        fun staticMethod(name: String, age: Int): String {
            return name
        }
    }


}